dynatrace/src/main/java/com/dynatrace/opentelemetry/metric/MetricAdapter.java

```diff
@@ -190,6 +246,24 @@ private static void addGaugeData(
     }
   }
 
+  /**
+   * This function first transforms the given labels to dimensions using {@link
+   * #convertLabelsToDimensions(Labels)} and then adds static labels that are stored in the
+   * singleton instance, if there are any.
+   *
+   * @param labels the labels to be transformed by {@link #convertLabelsToDimensions}.
+   * @return A list of {@link Dimension} objects, that will be serialized.
+   */
+  private static List<Dimension> getCombinedDimensions(Labels labels) {
+    List<Dimension> dynamicDimensions = convertLabelsToDimensions(labels);
+
+    if (getInstance().constantDimensions != null) {
+      dynamicDimensions.addAll(getInstance().constantDimensions);
```

Comment:
This will add both dimensions instead of overwriting them in case there are duplicate keys.

Reply:
Please add a test covering that as well :-)